cluster-autoscaler/simulator/cluster.go

```diff
@@ -73,6 +73,10 @@ const (
 	NotUnreadyLongEnough
 	// NodeGroupMinSizeReached - node can't be removed because its node group is at its minimal size already.
 	NodeGroupMinSizeReached
+	// NodeGroupMaxDeletionCountReached - node can't be removed because max node count to be removed value set in planner reached
+	NodeGroupMaxDeletionCountReached
+	// AtomicScaleDownFailed - node can't be removed as node group has ZeroOrMaxNodeScaling enabled and number of nodes to remove are not equal to target size
+	AtomicScaleDownFailed
```

Comment:
Won't this break atomic scale down completely? Consider:
1. Two nodes A & B have to be deleted atomically together.
2. A becomes empty, but something still runs on B.
3. Scale down decides A is unremovable since it wouldn't be an atomic scale down.
4. B becomes empty.
5. Following iteration A is filtered out from unneeded nodes since it was unremovable.
6. B is marked as unremovable because it cannot be atomically removed with unremovable node A.
7. ...the pattern repeats, with A & B alternating in unremovable state.

Reply:
For 5th point here, I think A will not be filtered out
A will be added to unremovableNode but won't have a TTL so it wont be blocked from being unneeded 

We [add timeout](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/core/scaledown/planner/planner.go#L299) to nodes with [blocking pods only ](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/core/scaledown/planner/planner.go#L283) and others marked as unremovable are [just used for monitoring](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/core/static_autoscaler.go#L640)

So I think node A will still be considered for removal for second iteration
If this make sense to you I can test such scenario and double check if these nodes will be blocked from scale down or not